### ISSUE TITLE
Feature suggestion : sources

### ISSUE THREAD

**Issue by raphink** [eljeffeg/SmartCopy#59]
Currently SmartCopy can add URL references to the about section of profiles.

It would be even better if it could add a document with the URL and tag sources in it for all the data imported from the URL.

**Comment by raphink** [comment]
This would document imports even better and would allow to see exactly what info was imported from a given URL. Ideally, SmartCopy would find existing documents with the same URL if they exist.

**Comment by raphink** [comment]
Another option could be for Geni to expose the Matches API so we can add these URLs as matches, although that in itself will not document the sources (which I still believe would be useful).

**Comment by jeffg2k** [comment]
Yeah - It would be nice if we could add it as a match and it would be nice if we could link sources to matches.  But I would only be interested in linking sources to record matches, not trees.  Same goes for documents.  I don't see online trees as a sources themselves, though the online tree could reference sources.  I would only want to add a source if it referenced actual documents or authorities.

**Comment by jeffg2k** [comment]
I've asked in the past for access to the Match stuff, but it's integrated with MyHeritage.  Not sure how complicated that would be for them based on the various sites SC may use.

**Comment by raphink** [comment]
I see. I agree about the sources. Sources are supposed to be dated and localized, so it doesn't make sense to use a tree for that. But it could make sense to use that for cemetery websites and such, or even archive ones.

As for matches, there could be lots of use cases, e.g. Wikipedia articles, etc.

**Comment by stbodie** [comment]
Revisiting this with a much smaller scope than the original discussion, which stalled on whether tree URLs even count as "sources" in Geni's own formal citations schema - that's a bigger product/API question this issue was never going to resolve on its own.

Instead: reuse the existing "* Reference: [url recordtype] - [SmartCopy]: timestamp" note that already gets appended to \`about_me\` on every submission (this has worked reliably for years). Append a short, category-level summary of what that submission actually touched, e.g.:

\`\`\`
* Reference: [https://www.ancestry.com/... Ancestry Genealogy] - [https://www.geni.com/projects/SmartCopy/18783 SmartCopy]: ''Aug 15 2026, 20:52:50 UTC'' (updated: photo, name, gender, living status, birth)
\`\`\`

Category-level (not field-by-field) so it stays skimmable even on a large multi-field update, and doesn't turn \`about_me\` into a growing changelog. Implemented in 2ca541a.

This doesn't solve the full original ask (formal sources/citations tied to Geni Matches), but it does address the underlying goal - documenting what was actually copied and from where, visible right on the profile - without needing any changes to Geni's own data model. Closing this out on that basis; if the fuller formal-sources idea is still wanted, it'd need its own issue with the product question (what counts as a source) resolved first.
